src/components/MyAccount/Settings/AccountSettingsTab.tsx

```diff
@@ -34,56 +34,54 @@ const AccountSettingsTab = ({ settingsData }: { settingsData: Patron }) => {
 
   const submitAccountSettings = async (e) => {
     e.preventDefault()
+    setIsLoading(true)
     const payload = parsePayload(e.target, mostRecentPatronData)
-    try {
-      const response = await fetch(
-        `${BASE_URL}/api/account/settings/${mostRecentPatronData.id}`,
-        {
-          method: "PUT",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify(payload),
-        }
-      )
-      const responseData = await response.json()
-      if (response.ok) {
-        setMostRecentPatronData(updatePatronData(mostRecentPatronData, payload))
-        setCurrentlyEditing(false)
-        setModalProps(successModalProps)
-        openModal()
-      } else {
-        alert(`error: ${responseData}`)
-        setModalProps(failureModalProps)
-        openModal()
+    const response = await fetch(
+      `/research/research-catalog/api/account/settings/${mostRecentPatronData.id}`,
+      {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(payload),
       }
```

Comment:
Very clever way to put this all together (38-46) this is by far the most complicated PUT so I'm glad it's been separated out